activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -118,9 +118,9 @@ def dangerous_class_method?(method_name)
         RESTRICTED_CLASS_METHODS.include?(method_name.to_s) || class_method_defined_within?(method_name, Base)
       end
 
-      def class_method_defined_within?(name, klass, superklass = klass.superclass) # :nodoc:
+      def class_method_defined_within?(name, klass) # :nodoc:
```

Comment:
Actually, this method is only ever called with `Base`, so this simplifies further by removing the `klass` argument and substituting `klass` for `Base`. But maybe the intention was to follow the form of `method_defined_within?` ?

But I think if the method is only ever used to check one thing, and it's private, then t should be defined in terms of that one thing...